tutor/src/components/annotations/annotation.jsx

```diff
@@ -147,78 +140,118 @@ export default class AnnotationWidget extends React.Component {
 
   @observable activeHighlight = null;
   @observable widgetStyle = null;
-  @observable needToGetReferenceElements = true;
+  @observable scrollToPendingAnnotation;
   @computed get showWindowShade() {
     return User.annotations.ux.isSummaryVisible;
   }
-
+  @observable canRenderSidebarButtons = false;
   @observable parentRect = {};
   @observable referenceElements = [];
 
   @computed get annotationsForThisPage() {
     return this.allAnnotationsForThisBook.filter(item =>
       (item.selection.chapter === this.props.chapter) &&
-      (item.selection.section === this.props.section) &&
-      this.referenceElements.find((el) => el.id === item.selection.elementId)
+                                                       (item.selection.section === this.props.section) &&
+                                                       this.referenceElements.find((el) => el.id === item.selection.elementId)
     );
   }
 
   @computed get allAnnotationsForThisBook() {
     return filter(User.annotations.array, { courseId: this.props.courseId });
   }
 
+  setupPendingHighlightScroll(windowHash) {
+    const highlightMatch = windowHash.match(/highlight-(.*)/);
+    if (!highlightMatch) { return; }
+    this.scrollToPendingAnnotation = () => {
+      const id = highlightMatch[1];
+      const annotation = User.annotations.get(id);
+      if (annotation) {
+        const selection = this.restoreSelectionWithReferenceId(annotation.selection);
+        scrollToSelection(this.props.windowImpl, selection);
+      } else {
+        Logging.error(`Page attempted to scroll to annotation id '${id}' but it was not found`);
+      }
+      this.scrollToPendingAnnotation = null;
+    };
+  }
+
   componentDidMount() {
+    if (this.props.windowImpl.location.hash) {
+      this.setupPendingHighlightScroll(this.props.windowImpl.location.hash);
+    }
+
     this.handleSelectionChange = debounce(() => {
       if (!this.activeHighlight) {
         this.setWidgetStyle();
       }
     }, 80);
-    this.props.windowImpl.document.addEventListener('selectionchange', this.handleSelectionChange);
+
+    when(
+      () => User.annotations.any,
```

Comment:
Does this mean that if the user has no annotations, the user will not be able to create any?

Reply:
yep, that's the case.  Glad you spotted this.  I think this can check `User.api.isPending`  instead. I'll test that